astropy/cosmology/parameter/tests/test_parameter.py

```diff
@@ -196,16 +188,16 @@ def test_Parameter_listed(self, cosmo_cls, all_parameter):
         assert isinstance(all_parameter, Parameter)
 
         # the reverse: check that if it is a Parameter, it's listed.
-        # note have to check the more inclusive ``__all_parameters__``
-        assert all_parameter.name in cosmo_cls.__all_parameters__
-        if not all_parameter.derived:
-            assert all_parameter.name in cosmo_cls.__parameters__
+        if all_parameter.derived:
+            assert all_parameter.name in cosmo_cls.derived_parameters
+        else:
+            assert all_parameter.name in cosmo_cls.parameters
 
     def test_parameter_related_attributes_on_Cosmology(self, cosmo_cls):
         """Test `astropy.cosmology.Parameter`-related on Cosmology."""
         # establish has expected attribute
-        assert hasattr(cosmo_cls, "__parameters__")
-        assert hasattr(cosmo_cls, "__all_parameters__")
+        assert hasattr(cosmo_cls, "_parameters_")
```

Comment:
Here, you are testing implementation rather than API, which I think is not needed.

Reply:
As in, scrap the whole ``test_parameter_related_attributes_on_Cosmology``.

Reply:
At some level, yes. But obviously more a thought than a big deal.

Reply:
I agree. It's largely superfluous

Reply:
Done.